src/sql/workbench/browser/modal/modal.ts

```diff
@@ -566,20 +630,37 @@ export abstract class Modal extends Disposable implements IThemable {
 	 * Called by the theme registry on theme change to style the component
 	 */
 	public style(styles: IModalDialogStyles): void {
-		this._dialogForeground = styles.dialogForeground;
+		this._dialogForeground = styles.dialogForeground ? styles.dialogForeground : this.getThemeColor(editorWidgetForeground);
+
 		this._dialogBorder = styles.dialogBorder;
-		this._dialogHeaderAndFooterBackground = styles.dialogHeaderAndFooterBackground;
-		this._dialogBodyBackground = styles.dialogBodyBackground;
+
+		if (this._modalOptions.dialogStyle === 'callout' || this._modalOptions.dialogStyle === 'calloutCompact') {
+			this._dialogHeaderAndFooterBackground = styles.dialogBodyBackground ? styles.dialogBodyBackground : this.getThemeColor(SIDE_BAR_BACKGROUND);
+
+		} else {
+			this._dialogHeaderAndFooterBackground = styles.dialogHeaderAndFooterBackground ? styles.dialogHeaderAndFooterBackground : this.getThemeColor(SIDE_BAR_BACKGROUND);
+		}
+
+		this._dialogBodyBackground = styles.dialogBodyBackground ? styles.dialogBodyBackground : this.getThemeColor(editorBackground);
+
 		this.applyStyles();
 	}
 
+	public getThemeColor(themeColorName: string): Color {
+		let themeColor: Color;
+		registerThemingParticipant((theme) => {
```

Comment:
I'm not sure what this is supposed to be doing - you pass in a callback to set the themeColor but then immediately return. I wouldn't expect this is actually return the expected color as it is. 

Typically for theming you should be doing two things : 

1. Using themeService.getColorTheme() to get the current theme and then use that to set the colors/styles
1. Registering a theme participant to then update those colors/styles when the theme changes